### PR TITLE
ISM Template Migration

### PR DESCRIPTION
**Pull request by thalurur**
Signed-off-by: Ravi Thaluru <thalurur@users.noreply.github.com>

*Issue #, if available:*
N/A

*Description of changes:*
* Porting the template migration code
* Using metadataservice status setting to determine if the metadata migration is enabled rather than enabled setting
* Porting other minor missing bugfixes

*CheckList:*
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/index-management/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/migration/MigrationServices.kt [hunk 1]
```diff
@@ -0,0 +1,364 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * com.maddyhome.idea.copyright.pattern.CommentInfo@7a1a92a5
```
Comment (bowenlan-amzn) [inline src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/migration/MigrationServices.kt:5]: What is this?

File: src/test/kotlin/org/opensearch/indexmanagement/indexstatemanagement/migration/MigrationServicesIT.kt [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
```
Comment (bowenlan-amzn) [inline src/test/kotlin/org/opensearch/indexmanagement/indexstatemanagement/migration/MigrationServicesIT.kt:14]: We may want this to be our opensearch license header

File: src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/migration/MigrationServices.kt (hunk 1 above)
Comment (thalurur) [reply src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/migration/MigrationServices.kt:5]: will clean it up - I've been noticing that when I create a new file in my IDE the message is being added to header, will check on my end to disable this

File: src/test/kotlin/org/opensearch/indexmanagement/indexstatemanagement/migration/MigrationServicesIT.kt (hunk 2 above)
Comment (thalurur) [reply src/test/kotlin/org/opensearch/indexmanagement/indexstatemanagement/migration/MigrationServicesIT.kt:14]: will do

**Review by bowenlan-amzn — APPROVED** [review]
LGTM, thanks.

**Review by downsrob — APPROVED** [review]

**Pull request merged**